tracing-opentelemetry-http/src/main/java/io/micronaut/tracing/opentelemetry/instrument/http/client/OpenTelemetryClientFilter.java

```diff
@@ -0,0 +1,119 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.tracing.opentelemetry.instrument.http.client;
+
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.async.publisher.Publishers;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.MutableHttpRequest;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.filter.ClientFilterChain;
+import io.micronaut.http.filter.HttpClientFilter;
+import io.micronaut.tracing.opentelemetry.instrument.http.AbstractOpenTelemetryFilter;
+import io.micronaut.tracing.opentelemetry.instrument.util.OpenTelemetryExclusionsConfiguration;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import jakarta.inject.Named;
+import org.jetbrains.annotations.NotNull;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscription;
+import reactor.core.CoreSubscriber;
+
+import static io.micronaut.tracing.opentelemetry.instrument.http.client.OpenTelemetryClientFilter.CLIENT_PATH;
+
+/**
+ * An HTTP client instrumentation filter that uses Open Telemetry.
+ *
+ * @author Nemanja Mikic
+ * @since 4.1.0
+ */
+@Filter(CLIENT_PATH)
+public class OpenTelemetryClientFilter extends AbstractOpenTelemetryFilter implements HttpClientFilter {
+
+    private final Instrumenter<MutableHttpRequest<?>, HttpResponse<?>> instrumenter;
+
+    /**
+     * Initialize the open tracing client filter with tracer and exclusion configuration.
+     *
+     * @param exclusionsConfig The {@link OpenTelemetryExclusionsConfiguration}
+     * @param instrumenter The {@link OpenTelemetryHttpClientConfig}
+     */
+    public OpenTelemetryClientFilter(@Nullable OpenTelemetryExclusionsConfiguration exclusionsConfig, @Named("micronautHttpClientTelemetryInstrumenter") Instrumenter<MutableHttpRequest<?>, HttpResponse<?>> instrumenter) {
+        super(exclusionsConfig == null ? null : exclusionsConfig.exclusionTest());
+        this.instrumenter = instrumenter;
+    }
+
+    @Override
+    public Publisher<? extends HttpResponse<?>> doFilter(MutableHttpRequest<?> request,
+                                                         ClientFilterChain chain) {
+
+        Publisher<? extends HttpResponse<?>> requestPublisher = chain.proceed(request);
+
+        if (shouldExclude(request.getPath())) {
+            return requestPublisher;
+        }
+
+        Context parentContext = Context.current();
+        if (!instrumenter.shouldStart(parentContext, request)) {
+            return requestPublisher;
+        }
+
+        return (Publishers.MicronautPublisher<HttpResponse<?>>) actual -> {
```

Comment:
I might have made a mistake in the last review, subscriber doesn't need to be `CoreSubscriber` but publisher needs to be `MicronautPublisher`+`CorePublisher` and delegate context if actual is also `CorePublisher` check `micronaut-tracing` for examples